tests/virt_autotest/login_console.pm

```diff
@@ -128,7 +128,18 @@ sub login_to_console {
             send_key 'down';
             send_key 'ret';
             #wait sshd up
-            assert_screen('sshd-server-started', 180);
+            if (!check_screen('sshd-server-started', 180)) {
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

I am convinced this approach is not good and should be avoided

Reply:
To be honest I can not use assert_screen here. This pull request aims to solve the unnecessary failure. Introducing assert_screen back with more needle only make the current test run worse.

Reply:
If you insist, I think I have to drop any "check or assert screen". Use check_ports only.

Reply:
Yes, that can be an option. So either assert_screen with all the options or no needle check at all

Reply:
Updated